fw/sock.c

```diff
@@ -1047,6 +1097,17 @@ ss_tcp_state_change(struct sock *sk)
 		 * done after all pending data has been sent.
 		 */
 		ss_close(sk, SS_F_SYNC);
+	} else if (sk->sk_state == TCP_FIN_WAIT2) {
```

Comment:
Looks like `TCP_FIN_WAIT2` is not enough, if closing has started simultaneously state will be changed to `TCP_CLOSING` that not handled in our case. Thus, `ss_linkerror` will not be called.